### PR TITLE
ASRC-199: Authenticate administrative users

### PR DESCRIPTION
**Pull request by dgtombs**
Add an in-memory user store of administrative users in addition to the `VistaUserDetailsService` that loads users from VistA.

Also add a user-friendly access denied page. I had to refactor the way the global SrcalcInfo object is made available to the pages to also make it available to `accessDenied.jsp`.

### CODE REVIEW

File: srcalc/src/main/java/gov/va/med/srcalc/security/VistaUserDetails.java [hunk 1]
```diff
@@ -20,26 +20,23 @@
...
+    public VistaUserDetails(final VistaPerson vistaPerson)
     {
         fVistaPerson = vistaPerson;
-        fAuthorities = authorities;
     }
 
+    /**
+     * This is simple: all VistA users are treated alike, so this method always
+     * returns a one-element collection containing {@link Roles#ROLE_USER}.
+     */
```
Comment (jamesbigmac) [inline srcalc/src/main/java/gov/va/med/srcalc/security/VistaUserDetails.java]: Will we need a class like VistaAdminDetails for a VistA admin, or is there no such thing?

File: srcalc/src/main/java/gov/va/med/srcalc/web/AppAttributesInitializer.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    
+    /**
+     * Adds the attributes to the {@link ServletContext}.
+     */
+    @Override
+    public void contextInitialized(final ServletContextEvent event)
+    {
+        final ServletContext servletContext = event.getServletContext();
+        final SrcalcInfo srcalcInfo = WebUtils.readSrcalcInfo(servletContext);
+        servletContext.setAttribute(ATTRIBUTE_APP_INFO, srcalcInfo);
```
Comment (jamesbigmac) [inline srcalc/src/main/java/gov/va/med/srcalc/web/AppAttributesInitializer.java]: Why do we get information that was already there, then put it back into the servlet context? Is this to make it easier to access the implementation title and version?

File: srcalc/src/test/java/gov/va/med/srcalc/web/WebConfigIT.java [hunk 3]
```diff
@@ -48,9 +40,5 @@ public static void setupJndi() throws NamingException
     public void testWebConfig()
```
Comment (jamesbigmac) [inline srcalc/src/test/java/gov/va/med/srcalc/web/WebConfigIT.java]: Is there a way to make this test more useful, or is just loading the application context good enough?

Comment (dgtombs) [inline srcalc/src/test/java/gov/va/med/srcalc/web/WebConfigIT.java]: Done.

**Pull request merged**